### PR TITLE
Combine summary and non-summary pipelines

### LINKED ISSUE #38: Combine Dataflow pipelines
For context, we have an existing Dataflow pipeline ([`bigquery_import.py`](https://github.com/HTTPArchive/bigquery/blob/master/dataflow/python/bigquery_import.py)) that generates the pages, requests, lighthouse, and response_bodies tables. This pipeline runs at the end of the crawl and processes all HARs at once. In #16 we're creating a new Dataflow pipeline to generate the summary_pages and summary_requests tables. This pipeline runs incrementally whenever a HAR is available for processing, even while other pages are still being tested.

Combine these two Dataflow pipelines so that we have a single streaming pipeline that generates the data for all tables one page (HAR) at a time.

### PR DESCRIPTION
**Pull request by giancarloaf**
Fixes #38

- [x] Import [bigquery_import.py](https://github.com/HTTPArchive/bigquery/blob/3dfb3460a80231ffbe1baa8c3f7e902a1b6b188b/dataflow/python/bigquery_import.py) with git history
- [x] refactor bigquery_import.py and import_har.py to reuse steps across multiple pipelines
- [x] create `combined_pipeline.py`
- [x] validate with small test runs
- [x] create new or import existing run scripts and documentation from [httparchive/bigquery](https://github.com/HTTPArchive/bigquery)
- [ ] validate with full test (e.g. April 1 2022)

**Validation**
combined job - https://console.cloud.google.com/dataflow/jobs/us-west1/2022-06-25_20_50_23-159409301539775907;step=?project=httparchive

non-summary job - https://console.cloud.google.com/dataflow/jobs/us-west1/2022-06-25_20_56_29-7795573434157386448;step=?project=httparchive

summary job - https://console.cloud.google.com/dataflow/jobs/us-west1/2022-06-25_20_55_30-3035437626606559196?project=httparchive

### CODE REVIEW

File: modules/non_summary_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+        )
+        return None
+
+    entries = har.get("log").get("entries")
+
+    requests = []
+
+    for request in entries:
+
+        request_url = request.get("_full_url")
```
Comment (rviscomi) [inline modules/non_summary_pipeline.py:133]: Same suggestion from the `all` pipeline
```suggestion
        request_url = request.get("_full_url")
        
        if not request_url:
            logging.warning('Skipping empty request URL for "%s"', page_url)
            continue
```

**Review by rviscomi — APPROVED** [review]

**Comment by rviscomi** [comment]
Looks like there's also one small linting error to fix, then we should be good

File: modules/combined_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+        group = parser.add_mutually_exclusive_group(required=True)
+        group.add_argument(
+            "--input",
+            help="Input file glob to process. Example: gs://httparchive/crawls/*Jan_1_2022",
+        )
+
+        group.add_argument(
+            '--input_file',
+            help="Input file containing a list of HAR files. Example: gs://httparchive/crawls_manifest/android-May_12_2022.txt"
```
Comment (rviscomi) [inline modules/combined_pipeline.py:35]:
```suggestion
            help="Input file containing a list of HAR files. " + \
                "Example: gs://httparchive/crawls_manifest/android-May_12_2022.txt"
```

**Comment by rviscomi** [comment]
Go home, linter, you're drunk

image

image

**Comment by tunetheweb** [comment]
Two different linters!

editorconfig wants multiples of 4 spaces for python
flake8 allows hanging indents (even if uneven number of indents to do that).

However, flake8 doesn't _insist_ on hanging indents. So you can please both by not using them. And in fact black doesn't like hanging indents so would have auto corrected.

Will submit a PR so we don't have to force merge in future.

**Comment by tunetheweb** [comment]
Fixed in #108

**Pull request merged**